Add flag to enable mutual certificate authentication for old Akka Remote SSL transport

jrudolph opened the pull request:
/cc @wsargent

jrudolph committed:
=rem #13874 further cleanup of SSLSettings / NettySSLSupport

jrudolph committed:
+rem #13874 allow requiring mutual authentication for old akka remote ssl transport

patriknw commented on akka-remote/src/main/scala/akka/remote/transport/netty/NettySSLSupport.scala:
```diff
@@ -34,25 +37,51 @@ private[akka] class SSLSettings(config: Config) {
+    } catch {
+      case e: FileNotFoundException    ⇒ throw new RemoteTransportException("Server SSL connection could not be established because key store could not be loaded", e)
+      case e: IOException              ⇒ throw new RemoteTransportException("Server SSL connection could not be established because: " + e.getMessage, e)
+      case e: GeneralSecurityException ⇒ throw new RemoteTransportException("Server SSL connection could not be established because SSL context could not be constructed", e)
```
should `Server` still be in these exception messages?

drewhk replied:
Good point. I think the message needs to be changed.

patriknw reviewed (approved):
LGTM

ktoso commented:
LGTM, can we first merge mine though? https://github.com/akka/akka/pull/21749 ;-)

drewhk commented:
Missing test? Shouldn't we have a test that fails if an untrusted system manages to connect?

ktoso commented:
Can you rebase @jrudolph sorry for that

ktoso commented:
LGTM :)

jrudolph commented:
Rebased. And I agree we should add tests before claiming that it actually works. Before I had changed the default to run it with the setting on, so that put more coverage on the new feature but we also need to test the bad case as well. I got stuck on understanding how akka-remote really works while writing the test. When it rejects the connection, you might only see that messages are not transmitted but not why. So, we need 1) test coverage for that feature 2) improve logging no to ignore the exceptions coming out of failed connection attempts.

ktoso commented:
Please open a ticket for the remaining work 1) and 2) here @jrudolph

jrudolph commented:
https://github.com/akka/akka/issues/21757
https://github.com/akka/akka/issues/21758
